vm/vm.go

```diff
@@ -117,7 +117,7 @@ func Call(contractAddr, selector *felt.Felt, calldata []felt.Felt, blockNumber,
 
 // Execute executes a given transaction set and returns the gas spent per transaction
 func Execute(txns []core.Transaction, declaredClasses []core.Class, blockNumber, blockTimestamp uint64,
-	sequencerAddress *felt.Felt, state core.StateReader, network utils.Network,
+	sequencerAddress *felt.Felt, state core.StateReader, network utils.Network, paidFeeOnL1 *felt.Felt,
```

Comment:
This is not possible for estimateMessageFee, but to keep `Execute` API suitable for all usecases of execution;

`txns` could contain multiple `L1Handlers`. Similar to `declaredClasses []core.Class` we need to have `paidFeesOnL1 []*felt.Felt`